resources/model/OpenStudio.idd

```diff
@@ -14186,6 +14186,341 @@ OS:HeatPump:PlantLoop:EIR:Cooling,
+   A12,\field Normalized Capacity Function of Temperature Curve Name
+       \required-field
+       \type object-list
+       \object-list BivariateFunctions
+       \note: CAPFT - Normalized Capacity Function of Temperature Curve Name,
+       \note which is a biquadratic curve or a lookup table.
+   A13,\field Fuel Energy Input Ratio Function of Temperature Curve Name
+       \required-field
+       \type object-list
+       \object-list BivariateFunctions
+       \note EIRFT - Fuel Energy Input Ratio Function of Temperature Curve Name,
+       \note which is a biquadratic curve or a lookup table.
+   A14,\field Fuel Energy Input Ratio Function of PLR Curve Name
+       \required-field
+       \type object-list
+       \object-list UnivariateFunctions
+       \note EIRFPLR - Fuel Energy Input Ratio Function of Part Load Ratio(PLR) Curve Name,
+       \note which is a cubic curve or a lookup table.
+   N7, \field Minimum Part Load Ratio
+       \minimum> 0.0
+       \maximum 1.0
+       \required-field
+       \note Minimum Part Load Ratio (PLR) in between 0 and 1
+   N8, \field Maximum Part Load Ratio
+       \minimum> 0.0
+       \maximum 1.0
+       \required-field
+       \note Maximum Part Load Ratio (PLR) in between 0 and 1
+   A15,\field Defrost Control Type
+       \type choice
+       \key Timed
+       \key OnDemand
+       \required-field
+       \note Defrost operation control type: Timed or OnDemand
+   N9, \field Defrost Operation Time Fraction
+       \minimum 0
+       \maximum 1
+       \required-field
+       \note Defrost operation time fraction, which will be used only for Timed Defrost Control Type.
+   A16,\field Fuel Energy Input Ratio Defrost Adjustment Curve Name
+       \type object-list
+       \object-list UnivariateFunctions
+       \note EIRDEFROST - Energy Input Ratio Defrost Adjustment Curve Name,
+       \note which is a cubic curve or a lookup table function of Outdoor Air Temperature.
+       \note This field is only used when OnDemand Defrost Control Type is selected.
+   N10,\field Resistive Defrost Heater Capacity
+       \units W
+       \minimum 0
+       \required-field
+       \note The heating power of electric resistive defrost heater. Only used for timed defrost control type.
+   N11,\field Maximum Outdoor Dry-bulb Temperature for Defrost Operation
+       \units C
+       \minimum 0
+       \maximum 10
+       \required-field
+       \note There will be no defrost operation when outdoor air is above this temperature.
+   A17,\field Cycling Ratio Factor Curve Name
+       \type object-list
+       \object-list UnivariateFunctions
+       \note Cycling Ratio Factor (CRF) Curve Name,
+       \note which is a cubic curve or a lookup table function of Cycling Ratio (defined as = PLR/PLRmin);
+       \note if left blank, no adjustment will be made.
+   N12,\field Nominal Auxiliary Electric Power
+       \units W
+       \minimum 0
+       \note Nominal Auxiliary Electric Power in [W]
```

Comment:
required-field, hard set to 0.0 in ctor, no?

Reply:
What I'm saying is that unless leaving it blank provides a different E+ behavior than putting it as 0, then I would make it required-field and harcode to zero.

In this case, E+ assumes it to be zero anyways, so it's safe to make it required-field: https://github.com/NREL/EnergyPlus/blob/27196146f19827e08dd8f7ec4059da0fc7c1793c/src/EnergyPlus/PlantLoopHeatPumpEIR.hh#L259